src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -967,9 +967,7 @@ void CompileBroker::init_compiler_sweeper_threads() {
 
   // Ensure any exceptions lead to vm_exit_during_initialization.
   EXCEPTION_MARK;
-#if !defined(ZERO)
-  assert(_c2_count > 0 || _c1_count > 0, "No compilers?");
-#endif // !ZERO
```

Comment:
I forgot why I removed that. Let me put it back and run it through testing again.

Reply:
So the reason for that was an awkward possibility of ending up with 0 compilers. That was because during the argument check in ```CompilerConfig::check_args_consistency()``` We checked for ```Arguments::is_interpreter_only()``` instead  of ```CompilerConfig::is_interpreter_only()```. The former doesn't check if ```TieredStopAtLevel=0``` is present. Anyways, I fixed the check and reverted this change.